### PR TITLE
Don't box signal callbacks twice

### PR DESCRIPTION
**Pull request by sdroege**
And also change various callbacks to take a generic &Self as first
argument just because we can.

### CODE REVIEW

File: src/signal.rs [hunk 1]
```diff
@@ -206,66 +202,64 @@ mod spin_button {
...
     }
 
     unsafe extern "C" fn input_trampoline(this: *mut GtkSpinButton,
                                           new_value: *mut c_double,
-                                          f: &&(Fn(&SpinButton) -> Option<Result<f64, ()>> + 'static))
-                                          -> c_int {
-        match f(&from_glib_borrow(this)) {
+                                          f: &F)
+                                          -> c_int
+    T: IsA<SpinButton> {
```
Comment (EPashkin) [inline src/signal.rs]: No `where` ?

Comment (sdroege) [reply src/signal.rs]: Yeah I can't compile things currently until @GuillaumeGomez is done with his regen PRs :)

Comment (EPashkin) [reply src/signal.rs]: *3 and many other errors,

Comment (sdroege) [reply src/signal.rs]: Yeah we can fix and merge this one once the other things are done :)

**Comment by EPashkin** [comment]
Canceled CI

**Pull request merged**